### PR TITLE
feat: add ignore errors flags

### LINKED ISSUE #203: [REQ] flag to continue when there are errors
### What kind of request is this?

New feature

### What is your request or suggestion?

today, we fail patching if any package fails to patch

if patching fails for a package for whatever reason, we might want to continue anyway to produce an image

### PR DESCRIPTION
**Pull request by anubhav06**
- Added a new `--ignore-errors` flag
-  When this flag is used, even if we get an error while patching any of the packages, copa will still continue to produce an image.

Closes #203 

## How to test?

1. `trivy image --vuln-type os --ignore-unfixed -f json -o nginx.1.21.6.json mcr.microsoft.com/oss/nginx/nginx:1.21.6`
2.  First, run copa without the `--ignore-errors` flag:
```bash
sudo copa patch -i mcr.microsoft.com/oss/nginx/nginx:1.21.6 -r nginx.1.21.6.json -t 1.21.6-patched
```

image
image

3. Now run with the `--ignore-errors` flag enabled:
```bash
sudo copa patch -i mcr.microsoft.com/oss/nginx/nginx:1.21.6 -r nginx.1.21.6.json -t 1.21.6-patched --ignore-errors
```

image
image
image

We can see that despite the 2 errors, copa ignores them, and still produces an image

4. Verified, that except those 2 packages, rest of the packages are patched:

image

### CODE REVIEW

File: pkg/patch/cmd.go [hunk 1]
```diff
@@ -38,7 +39,8 @@ func NewPatchCmd() *cobra.Command {
 				ua.appImage,
 				ua.reportFile,
 				ua.patchedTag,
-				ua.workingFolder)
+				ua.workingFolder,
+				ua.ignoreError)
```
Comment (sozercan) [inline pkg/patch/cmd.go:43]: please add a test for this if possible

Comment (sozercan) [reply pkg/patch/cmd.go:43]: applies all the changes below too

Comment (anubhav06) [reply pkg/patch/cmd.go:47]: added in integration tests ([here](https://github.com/project-copacetic/copacetic/pull/247/files#diff-aa5a6d1a6e5802fb07e773e37a06b714eea384d4150f2691586196f3019f3713R61-R72))

File: integration/fixtures/test-images.json [hunk 2]
```diff
@@ -4,83 +4,103 @@
...
+        "description": "Valid rpm DB, yum present",
+        "ignoreErrors": false
+    },
+    {
+        "image": "mcr.microsoft.com/oss/nginx/nginx",
+        "tag" : "1.21.6",
+        "digest": "sha256:bcdc041d100a3cc9fec472a8133ad566166e366241587ff92e8cc8fcf955229d",
+        "distro": "Debian",
+        "description": "Valid dpkg/status, apt present",
+        "ignoreErrors": true
```
Comment (sozercan) [inline integration/fixtures/test-images.json:104]: does this fail to patch?

Comment (anubhav06) [reply integration/fixtures/test-images.json:104]: yes, it fails to patch without the `ignore-errors` flag.
See the [PR description](https://github.com/project-copacetic/copacetic/pull/247#issue-1847850011) (How to Test > 2nd Step), I've also included a screenshot of it

Comment (sozercan) [reply integration/fixtures/test-images.json:104]: @anubhav06 i can't replicate this one in main branch

`docker.io/grafana/grafana-image-renderer:3.4.0` has an error with libcryto and libssl

Comment (anubhav06) [reply integration/fixtures/test-images.json:104]: added

File: integration/fixtures/test-images.json [hunk 3]
```diff
@@ -4,83 +4,103 @@
...
-        "description": "Valid rpm DB, yum present"
+        "description": "Valid rpm DB, yum present",
+        "ignoreErrors": false
+    },
+    {
+        "image": "docker.io/grafana/grafana-image-renderer",
+        "tag" : "3.4.0",
+        "digest": "sha256:205a39f5b58f96b9ff81a0b523a60c26c86e88e76575696fcd6debde9de02197",
+        "distro": "Alpine",
+        "description": "Valid apk/db, apk present",
```
Comment (sozercan) [inline integration/fixtures/test-images.json:103]:
```suggestion
        "description": "Valid apk/db, apk present, fail to patch libssl/libcryto",
```

File: integration/fixtures/test-images.json (hunk 2 above)
Comment (sozercan) [reply integration/fixtures/test-images.json:104]: I am surprised this is not failing in the CI

**Review by sozercan — APPROVED** [review]
Thank you! LGTM

**Pull request merged**